File: modules/wem-admin-ui/src/main/resources/web/admin/common/js/schema/content/ContentTypeDeletedEvent.ts
module api.schema.content {

    export class ContentTypeDeletedEvent extends api.event.Event {

        private name: ContentTypeName;

        constructor(name: ContentTypeName) {
            super();
            this.name = name;
        }

        getContentTypeName(): ContentTypeName {
            return this.name;
        }

        static on(handler: (event: ContentTypeDeletedEvent) => void) {
            api.event.Event.bind(api.util.getFullName(this), handler);
        }

        static un(handler?: (event: ContentTypeDeletedEvent) => void) {
            api.event.Event.unbind(api.util.getFullName(this), handler);
        }


        static fromJson(json: ContentTypeDeletedEventJson): ContentTypeDeletedEvent {
            return new ContentTypeDeletedEvent(new ContentTypeName(json.name));
        }
    }

    export interface ContentTypeDeletedEventJson {
        name: string;
    }
}

File: modules/core-api/src/main/java/com/enonic/wem/api/schema/content/ContentTypeDeletedEvent.java
package com.enonic.wem.api.schema.content;

import com.enonic.wem.api.event.Event;

public class ContentTypeDeletedEvent
    implements Event
{
    private final ContentTypeName name;

    public ContentTypeDeletedEvent( final ContentTypeName name )
    {
        this.name = name;
    }

    public ContentTypeName getName()
    {
        return name;
    }
}
